mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/query/SortConverter.java

```diff
@@ -55,7 +55,7 @@ private static FieldSortBuilder getSortClause(SearchQuery.Sort.SortClause clause
             case Size :
                 return SortBuilders.fieldSort(JsonMessageConstants.SIZE);
             case SentDate :
-                return SortBuilders.fieldSort(JsonMessageConstants.SENT_DATE);
+                return SortBuilders.fieldSort(JsonMessageConstants.DATE);
```

Comment:
This "back-up plan" on internal date is currently done when creating the field SentDate to be indexed:

```
 String sentDate = DateResolutionFormater.DATE_TIME_FOMATTER
        .format(headerCollection.getSentDate()
            .orElse(internalDate));
```

And sentDate is considered in HeaderCollection, to be the **Date** header.

IMO nothing is too wrong with ES. Of course we should validate with test... Apparently we miss tests on the back-up to internalDate.